iceoryx_posh/test/moduletests/test_roudi_config_json_file_provider.cpp

```diff
@@ -0,0 +1,402 @@
+                 "\t\t{\n"
+                 "\t\t\t\"no_mem\":[\n"
+                 "\t\t\t]\n"
+                 "\t\t}\n"
+                 "\t]\n"
+                 "}\n";
+    ASSERT_EQ(iox::roudi::RouDiConfigFileParseError::SEGMENT_WITHOUT_MEMPOOL, parseJson(str).get_error());
+}
+
+TEST_F(JsonConfig_Failure_test, Mempool_Without_Chunk_Size_test)
+{
+    char str[] = "{\n"
+                 "\t\"general\":{\n"
+                 "\t\t\"version\" : 1\n"
+                 "\t},"
+                 "\t\"segment\":[\n"
+                 "\t\t{\n"
+                 "\t\t\t\"mempool\":[\n"
+                 "\t\t\t\t{\n"
+                 "\t\t\t\t\t\"count\":10000\n"
+                 "\t\t\t\t}\n"
+                 "\t\t\t]\n"
+                 "\t\t}\n"
+                 "\t]\n"
+                 "}\n";
+    ASSERT_EQ(iox::roudi::RouDiConfigFileParseError::MEMPOOL_WITHOUT_CHUNK_SIZE, parseJson(str).get_error());
+}
+
+TEST_F(JsonConfig_Failure_test, Mempool_Without_Chunk_Size_Wrong_Type_test)
+{
+    char str[] = "{\n"
+                 "\t\"general\":{\n"
+                 "\t\t\"version\" : 1\n"
+                 "\t},"
+                 "\t\"segment\":[\n"
+                 "\t\t{\n"
+                 "\t\t\t\"mempool\":[\n"
+                 "\t\t\t\t{\n"
+                 "\t\t\t\t\t\"size\":\"\"\n"
+                 "\t\t\t\t\t\"count\":10000\n"
+                 "\t\t\t\t}\n"
+                 "\t\t\t]\n"
+                 "\t\t}\n"
+                 "\t]\n"
+                 "}\n";
+    ASSERT_EQ(iox::roudi::RouDiConfigFileParseError::MEMPOOL_WITHOUT_CHUNK_SIZE, parseJson(str).get_error());
+}
+
+
+TEST_F(JsonConfig_Failure_test, Mempool_Without_Chunk_Count_test)
+{
+    char str[] = "{\n"
+                 "\t\"general\":{\n"
+                 "\t\t\"version\" : 1\n"
+                 "\t},"
+                 "\t\"segment\":[\n"
+                 "\t\t{\n"
+                 "\t\t\t\"mempool\":[\n"
+                 "\t\t\t\t{\n"
+                 "\t\t\t\t\t\"size\":32\n"
+                 "\t\t\t\t}\n"
+                 "\t\t\t]\n"
+                 "\t\t}\n"
+                 "\t]\n"
+                 "}\n";
+    ASSERT_EQ(iox::roudi::RouDiConfigFileParseError::MEMPOOL_WITHOUT_CHUNK_COUNT, parseJson(str).get_error());
+}
+
+TEST_F(JsonConfig_Failure_test, Mempool_Without_Chunk_Count_Wrong_Type_test)
+{
+    char str[] = "{\n"
+                 "\t\"general\":{\n"
+                 "\t\t\"version\" : 1\n"
+                 "\t},"
+                 "\t\"segment\":[\n"
+                 "\t\t{\n"
+                 "\t\t\t\"mempool\":[\n"
+                 "\t\t\t\t{\n"
+                 "\t\t\t\t\t\"size\":10000\n"
+                 "\t\t\t\t\t\"count\":\"\"\n"
+                 "\t\t\t\t}\n"
+                 "\t\t\t]\n"
+                 "\t\t}\n"
+                 "\t]\n"
+                 "}\n";
+    ASSERT_EQ(iox::roudi::RouDiConfigFileParseError::MEMPOOL_WITHOUT_CHUNK_COUNT, parseJson(str).get_error());
+}
+
```

Comment:
A test that tries to exceed NUMBER_OF_JSON_NODES is missing.

Reply:
added